src/cli/commands/compile.rs

```diff
@@ -0,0 +1,29 @@
+use std::{fs::File, io::Write, path::PathBuf, process::Command};
+
+const JSON_FILE_EXTENTION: &str = "json";
+
+pub fn compile(program: &PathBuf) -> PathBuf {
```

Comment:
Once returned the path is not supposed to be altered. 
You can return a `Path` rather than a `PathBuf`

Reply:
You cannot return a `Path`?